eos-phone-home

```diff
@@ -88,74 +90,80 @@ class PhoneHome(object):
     MANDATORY_STRINGS = ('image', 'vendor', 'product', 'release', 'serial',
                          'metrics_environment')
 
-    def __init__(self, is_debug, force):
+    def __init__(self, is_debug, force, root=''):
         self._debug = is_debug
         self._force = force
+        self._root = root
         self._variables = {}
 
-        self._activated_path = os.path.join(self.STATE_DIRECTORY, 'activated')
-        self._count_path = os.path.join(self.STATE_DIRECTORY, 'count')
+        self._state_directory = root + '/var/lib/eos-phone-home'
```

Comment:
Hmm..maybe we should use os.path.join for all path concats?

Reply:
I did consider this, but rejected it because:

```python
In [2]: os.path.join('/tmp/foo', '/var/lib/eos-phone-home')
Out[2]: '/var/lib/eos-phone-home'
```

But now I think about it again, we could use `/` as the default value of `root` and use relative paths throughout. Good point.

Reply:
Pushed a fixup commit that does this, though a helper method.